storops/unity/calculator.py

```diff
@@ -295,6 +313,254 @@ def _sp_pct(path, curr, prev):
+        curr_part3 = curr.by_path(part3)
+
+        delta_part1 = curr_part1.sum_sp() - prev_part1.sum_sp()
+        delta_part2 = curr_part2.sum_sp() - prev_part2.sum_sp()
+        delta_part3 = curr_part3.sum_sp() - prev_part3.sum_sp()
+
+        ret = delta_part1 / (delta_part2 + delta_part3)
+
+    else:
+        ret = IdValues()
+
+    return ret
+
+
+@metric_calculator
+def lun_queue_length(path, prev, curr):
+    if len(path) != 3:
+        raise ValueError(
+            'takes in "currentIOCount", "busyTime" and '
+            '"idleTime" counter.')
+
+    part1, part2, part3 = path
+    if all_not_none(prev, curr):
+        prev_part1 = prev.by_path(part1)
+        prev_part2 = prev.by_path(part2)
+        prev_part3 = prev.by_path(part3)
+
+        curr_part1 = curr.by_path(part1)
+        curr_part2 = curr.by_path(part2)
+        curr_part3 = curr.by_path(part3)
+
+        delta_part1 = curr_part1.sum_sp() - prev_part1.sum_sp()
+        delta_part2 = curr_part2.sum_sp() - prev_part2.sum_sp()
+        delta_part3 = curr_part3.sum_sp() - prev_part3.sum_sp()
+
+        ret = delta_part1 * delta_part2 / (delta_part2 + delta_part3)
+
+    else:
+        ret = IdValues()
+
+    return ret
+
+
+@metric_calculator
+def sum_sp(path, _, curr):
+    path = only_one_path(path)
+    return curr.by_path(path).sum_sp()
+
+
+@metric_calculator
+def byte_rate(path, prev, curr):
+    if len(path) != 2:
+        raise ValueError('takes in "Blocks" and "blockSize" counter.')
+
+    part1, part2 = path
+    if all_not_none(prev, curr):
+        delta = curr.by_path(part1).combine_numeric_values(
+            curr.by_path(part2)) - prev.by_path(
+            part1).combine_numeric_values(
+            prev.by_path(part2))
+        ret = delta / curr.diff_seconds(prev)
+    else:
+        ret = IdValues()
+
+    return ret
+
+
+@metric_calculator
+def total_byte_rate(path, prev, curr):
+    if len(path) != 3:
+        raise ValueError(
+            'takes in "readBlocks" "writeBlocks" and '
+            '"blockSize" counter.')
+
+    part1, part2, part3 = path
+    return byte_rate((part1, part3), prev, curr) + byte_rate((part2, part3),
+                                                             prev, curr)
+
+
+@metric_calculator
+def sp_byte_rate(path, prev, curr):
+    if len(path) != 2:
+        raise ValueError('takes in "Blocks" and "blockSize" counter.')
+
+    part1, part2 = path
+    if all_not_none(prev, curr):
+        delta = curr.by_path(part1).combine_sp_values(
+            curr.by_path(part2)) - prev.by_path(part1).combine_sp_values(
+            prev.by_path(part2))
+        ret = delta / curr.diff_seconds(prev)
+    else:
+        ret = IdValues()
+
+    return ret
+
+
+@metric_calculator
+def sp_total_byte_rate(path, prev, curr):
+    if len(path) != 3:
+        raise ValueError(
+            'takes in "readBlocks" "writeBlocks" and '
+            '"blockSize" counter.')
+
+    part1, part2, part3 = path
```

Comment:
Better to name them as `readBlocks`, `writeBlocks` and `blockSize` if you are expecting them in the exact sequence.

Reply:
Done, already rename the variables.